### ISSUE TITLE
Disk control not correctly changing disks with BlueMSX

### ISSUE THREAD

**Issue by fragmental** [libretro/RetroArch#10044]
## Description

Whenever I try to change disks, when using the BlueMSX core, the disk does not change.  I've tried using an m3u then `Eject Disk`, changing the disk index, and then `Insert Disk`.  I've also tried `Load New Disk`.  Always the disk stays the same as the one that was initially loaded.

I don't know if this is a problem with retroarch or with the core, but I think I was able to get the disk to change somehow in 1.8.2, with BlueMSX.  (I only tried briefly with 1.8.2 before updating to 1.8.4 because I had read about the changes in the changelog.)  However,  the beetle saturn core seems to change discs without any issue.

Here is the relevant section of the log in debug verbosity
```[INFO] [Vulkan]: VSync => on
[INFO] Ejected virtual disk tray.
[INFO] [Vulkan]: VSync => on
[INFO] [Vulkan]: VSync => on
[INFO] Setting disk in tray: 2/3.
[INFO] Closed virtual disk tray.
[INFO] [Vulkan]: VSync => on
```

The vsync on message displays when I enter or leave the UI.

As you can see, the log seems to indicate that the disk has been changed, but that doesn't appear to be the case.

I've also created a git issue in the core repository in case it is a problem with the core https://github.com/libretro/blueMSX-libretro/issues/107

### Expected behavior

Disk should change

### Actual behavior

Disk appears to change in log, but does not change

### Steps to reproduce the bug

1. load up m3u for multidisk game (I use Aleste 2 to test)
2. try to change disk

### Bisect Results

This might have started happening with 1.8.4, but I'm not entirely certain

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.8.4 Git version f425d8dc8b
- BlueMSX: git aace4ae

### Environment information

- OS: Windows 10 1903 x64

**Comment by bslenul** [comment]
Works fine here, tested with Aleste 2 + RA 1.8.4 + same git version of BlueMSX, I load the .m3u file, wait for the message to switch disk, disk control > eject > switch index 2 > insert, press B on the controller then wait a bit for it to load.

https://www.youtube.com/watch?v=8bcIh7axJlA

My files and .m3u:

![image](https://user-images.githubusercontent.com/33353403/72935388-858fb380-3d65-11ea-97fa-df77cdddc059.png)

**Comment by fragmental** [comment]
Weird.  My setup is mostly the same.  The only major difference, from what I can tell, is that my ui is different.  I'm using the default blue interface.  https://i.imgur.com/SROU2Ml.png

**Comment by bslenul** [comment]
Hm, I see that you have State Slot -1 selected, is the game auto-loading from a state? Maybe that's the issue, make sure you have disk index 1 selected then quick menu > restart and give it another try.

MD5 of my disks also, if you want to compare game version:
* Disk 1: 883E46527DA292C148769425B30AD685
* Disk 2: 52B1591D7CD5921B8256E213A6859600
* Disk 3: F5BE2140761D3F3B36CF62598ED6004B

**Comment by fragmental** [comment]
Thinking that maybe it was a problem with XMB, I tried the other UI Drivers, and there was no change.

Using the exact same dsk files (verified with md5 sums), an m3u created with the exact same program, the exact same file names, the exact same UI driver (Ozone), the same save state slot, the same retroarch version, and the same bluemsx version it still won't change the disk.  I have no idea why.

**Comment by bslenul** [comment]
Not sure then :/ Post the full log, maybe there will be something helpful in it.

**Comment by fragmental** [comment]
Thanks for helping diagnose things.  Here's a full log https://drive.google.com/file/d/1J8zX5F20Mn7Tw9VpDso4gM0V_2q1VUGC/view?usp=sharing and here's the m3u https://drive.google.com/open?id=1VnPi1M0CZBRnQIlfPz1joZWqMxaGK4Y_ if anyone wants to try it.

**Comment by fragmental** [comment]
I tested this on a freshly updated retroarch, in Linux, on the same system via dual boot.  Sure enough, it changes as expected.  There must be something wrong with my retroarch in windows.  It could be something related to updating like an old config file or something.  I don't know.

**Comment by bslenul** [comment]
Could be, yeah! If it's a conflict with a previous setting you could try this:

* Rename your current `retroarch.cfg` to `retroarch_old.cfg` for example.
* Try loading the game again (RA will recreate a fresh new `retroarch.cfg`).
* If it still doesn't work then idk honestly :|
* But if it works you could upload both recreated `retroarch.cfg` and `retroarch_old.cfg` (remove twitch key and/or passwords inside if you have any). If we can find which option was conflicting that could be really helpful for people having this issue in the future :)

**Comment by fragmental** [comment]
OMG, that was it.  Here's my retroarch cfg https://drive.google.com/open?id=1A9Xd2WAmRYdWQUYSRnOMp5_d9JMaxZmn  I scrubbed my cheevos token from it, but otherwise it's unchanged.  I used winmerge to compare the new config, and nothing immediately jumped out at me as a potential cause of the problem.  There were several lines that didn't exist in the new config, though.

**Comment by fragmental** [comment]
Ok, I got it.  It's some conflict with the run-ahead.  If I swap disks with "RunAhead Use Second Instance" enabled it won't swap the disks.  However, using run-ahead with bluemsx without use second instance doesn't seem to work very well at all.  In order to be able to change disks, I have to leave RunAhead off before starting the disk, and leave it off.  If I turn it on at any point, I will not be able to change disks.  If there's only one disk swap that needs to be made, it can be turned off after that disk swap, but only if no other disks swapping needs to be done.

**Comment by fragmental** [comment]
I'm closing this issue, and I've created a new issue since I've diagnosed the bug. #10070
